File: lc_soln/0199_binary_tree_right_side_view/solution.py
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        if root is None:
            return result
        q = deque([root])
        while q:
            result.append(q[-1].val)
            n = len(q)
            for _ in range(n):
                node = q.popleft()
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
        return result


def main(): ...


if __name__ == "__main__":
    main()

File: lc_soln/0199_binary_tree_right_side_view/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
    public List<int> RightSideView(TreeNode root)
    {
        if (root is null)
        {
            return [];
        }
        List<int> result = [];
        var q = new LinkedList<TreeNode>();
        q.AddLast(root);
        while (q.Count > 0)
        {
            int n = q.Count;
            while (n-- > 0)
            {
                TreeNode node = q.First.Value;
                q.RemoveFirst();
                if (n == 0)
                {
                    result.Add(node.val);
                }
                if (node.left is not null)
                {
                    q.AddLast(node.left);
                }
                if (node.right is not null)
                {
                    q.AddLast(node.right);
                }
            }
        }
        return result;
    }
}
